### PR TITLE
[Keyboard]KPrepublic bm80v2 Keyboard ANSI support

### PR DESCRIPTION
**Pull request by edwardslau**
## Description
BM80 V2 ANSI layout 

## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm80v2/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kprepublic/bm80v2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2/rules.mk:21]:
```suggestion
NKRO_ENABLE = yes              # Enable N-Key Rollover
BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by edwardslau** [comment]
Modified the files as requested

File: keyboards/kprepublic/bm80v2/readme.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# bm80v2
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2/readme.md:1]: This needs to be filled out, RE:
https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md

**Review by drashna — APPROVED** [review]

File: keyboards/kprepublic/bm80v2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes        # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/kprepublic/bm80v2/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes         # Enable Bootmagic Light
```

**Review by tzarc — APPROVED** [review]

**Pull request merged**